khmer/kfile.py

```diff
@@ -167,3 +169,31 @@ def check_valid_file_exists(in_files):
         else:
             print('WARNING: Input file %s not found' %
                   in_file, file=sys.stderr)
+
+
+def add_output_compression_type(parser):
+    parser.add_argument('--gzip', default=False, action='store_true',
+                        help='Option to output as gzip')
+    parser.add_argument('--bzip', default=False, action='store_true',
+                        help='Option to output as bz2')
+
+
+def get_file_writer(file_handle, do_gzip, do_bzip):
+    ofile = None
+
+    if do_gzip and do_bzip:
```

Comment:
I think this can be handled at the option parsing stage by using mutually exclusive groups:

https://docs.python.org/2/library/argparse.html#mutual-exclusion